sound/soc/sof/sof-client.c

```diff
@@ -126,6 +127,28 @@ static inline int sof_register_ipc_msg_injector(struct snd_sof_dev *sdev)
 static inline void sof_unregister_ipc_msg_injector(struct snd_sof_dev *sdev) {}
 #endif /* CONFIG_SND_SOC_SOF_DEBUG_IPC_MSG_INJECTOR */
 
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_DEBUG_IPC_KERNEL_INJECTOR)
+static int sof_register_ipc_kernel_injector(struct snd_sof_dev *sdev)
+{
+	/* Only IPC3 supported right now */
+	if (sdev->pdata->ipc_type == SOF_IPC)
+		return sof_client_dev_register(sdev, "kernel_injector", 0, NULL, 0);
```

Comment:
nitpick: the reason that I suggested:
```c
if (sdev->pdata->ipc_type != SOF_IPC)
	return 0;

return sof_client_dev_register(sdev, "kernel_injector", 0, NULL, 0);
```
is that this way we can end up with shorter lines and aligned with how `sof_register_ipc_flood_test()` has been implemented.